### PR TITLE
rewrite kv breaker to take the comments above the kv block

### PR DESCRIPTION
**Pull request by XavierCHN**
it should be
```
    //=====================================
	// Comment
	//=====================================
	"Simayi_Yanjibo"
	{
		"AbilityTextureName"	"sima_yanjibo"
```
after break down

### CODE REVIEW

**Comment by XavierCHN** [comment]
changes about the spaces is automatic by visual studio. xD

File: D2Modkit/MainForm.cs [hunk 1]
```diff
@@ -1072,71 +1072,60 @@ private void breakUpBtn_Click(object sender, EventArgs e)
                         {
                             IEnumerable<KeyValue> kvs2 = kv.Children;
                             KeyValue[] kvArr = kvs2.ToArray();
+                            // record start line number and end line number of each Key-Value block
```
Comment (XavierCHN) [inline D2Modkit/MainForm.cs]: start of the change of the break method

File: D2Modkit/MainForm.cs [hunk 2]
```diff
@@ -1072,71 +1072,60 @@ private void breakUpBtn_Click(object sender, EventArgs e)
...
+                                StringBuilder sb = new StringBuilder();
+                                
+                                for (int p1 = startLineNumber[p]; p1 <= endLineNumber[p]; p1++)
+                                    sb.AppendLine(lines[p1]);
+                                File.WriteAllText(filePath,sb.ToString());
                             }
-
-                            // do the last entry.
-                            File.Create(filePath).Close();
-                            File.WriteAllText(filePath, entry);
```
Comment (XavierCHN) [inline D2Modkit/MainForm.cs]: end of the change about the break method

**Comment by Myll** [comment]
why'd you close? i will take a look at it soon

**Comment by XavierCHN** [comment]
Found a bug that will generate empty lines in some case and currently don't know why.
Will send it again right after fix it

**Comment by XavierCHN** [comment]
change is simple.
https://github.com/XavierCHN/Dota-2-ModKit/blob/master/D2Modkit/MainForm.cs#L1066

**Pull request merged**